atomicapp/nulecule/lib.py

```diff
@@ -94,7 +101,11 @@ def get_provider(self, provider_key=None, dry=False):
         # If provider_key isn't provided via CLI, let's grab it the configuration
         if provider_key is None:
             provider_key = self.config.get(GLOBAL_CONF)[PROVIDER_KEY]
-        provider_class = self.plugin.getProvider(provider_key)
+        try:
+            provider_class = self.plugin.getProvider(provider_key)
+        except NuleculeException as e:
+            logger.error(e)
+            sys.exit(1)
```

Comment:
Exiting here won't make much harm because no container has been pulled yet or no actual processing has been started, its just rendering and parsing of configuration files.

Correct me if I am wrong.